test/e2e/validation_test.go

```diff
@@ -299,19 +299,28 @@ func (tc *testContext) sshSetup() error {
 // runPowerShellSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed through
 // PowerShell, on the host specified by the provided IP.
 func (tc *testContext) runPowerShellSSHJob(name, command, ip string) (string, error) {
+	// Modify command to work when default shell is the newer Powershell version present on Windows Server 2022.
+	// We only test Windows Server 2022 if a custom VXLAN port is used (i.e. vSphere)
+	powershellDefaultCommand := command
+	if tc.hasCustomVXLAN {
+		powershellDefaultCommand = strings.ReplaceAll(command, "\\\"", "\"")
+	}
+
 	keyMountDir := "/private-key"
 	sshCommand := []string{"bash", "-c",
 		fmt.Sprintf(
 			// first determine if the host has PowerShell or cmd as the default shell by running a simple PowerShell
 			// command. If it succeeds, then the host's default shell is PowerShell
 			"if ssh -o StrictHostKeyChecking=no -i %s %s@%s 'Get-Help';"+
 				"then CMD_PREFIX=\"\";CMD_SUFFIX=\"\";"+
-				// if PowerShell is not the default shell, explicitly run the command through PowerShell
+				"COMMAND='{"+powershellDefaultCommand+"}';"+
+				// if PowerShell is not the default shell, explicitly run the unmodified command through PowerShell
 				"else CMD_PREFIX=\""+remotePowerShellCmdPrefix+" \\\"\";CMD_SUFFIX=\"\\\"\";"+
+				"COMMAND='{"+command+"}';"+
 				"fi;"+
-				"ssh -o StrictHostKeyChecking=no -i %s %s@%s ${CMD_PREFIX}' %s '${CMD_SUFFIX}",
+				"ssh -o StrictHostKeyChecking=no -i %s %s@%s ${CMD_PREFIX}\" & $COMMAND \"${CMD_SUFFIX}",
```

Comment:
i am mostly to blame but looking at this without this code fresh in my memory, this is very hard to understand

is this a logical AND? what is the function of the &?
this needs to be doc'd better.

Reply:
+1 can we have the reasoning behind this added to the commit msg as well?